x/metadata/keeper/query_server.go

```diff
@@ -73,10 +73,75 @@ func (k Keeper) GroupContext(c context.Context, req *types.GroupContextRequest)
 	return &types.GroupContextResponse{}, nil
 }
 
-// Record returns a collection of the records in a scope or a specific one by name
-func (k Keeper) Record(c context.Context, req *types.RecordRequest) (*types.RecordResponse, error) {
-	// TODO
-	return &types.RecordResponse{}, nil
+// RecordByScopeUUID returns a collection of the records in a scope or a specific one by name
+func (k Keeper) RecordByScopeUUID(c context.Context, req *types.RecordByScopeUUIDRequest) (*types.RecordByScopeUUIDResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	if req.GetScopeUuid() == "" {
+		return nil, status.Error(codes.InvalidArgument, "scope uuid cannot be empty")
+	}
+
+	uuid, err := uuid.Parse(req.GetScopeUuid())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid scope uuid: %s", err.Error())
+	}
+
+	scopeAddr := types.ScopeMetadataAddress(uuid)
+	records, err := k.record(c, &scopeAddr, req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.RecordByScopeUUIDResponse{ScopeUuid: req.GetScopeUuid(), ScopeId: scopeAddr.String(), Records: records}, nil
```

Comment:
Instead of calling `getscopeuuid` here  the local scope `uuid` variable can be used.  Using the name `uuid` for a local variable and the import reference is not a good idea though.

Reply:
🤦 Good catch!